bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/controller/MatterControllerClient.java

```diff
@@ -0,0 +1,329 @@
+                new Object[] { nodeId, connectionTimeoutMilliseconds }, connectionTimeoutMilliseconds / 1000 + 1);
+        return future.thenAccept(obj -> {
+            // Do nothing, just to complete the future
+        });
+    }
+
+    /**
+     * Request all cluster attribute data for the node from the controller, the actual data will be sent via a
+     * NodeDataListener event
+     * 
+     * @param nodeId
+     * @return
+     */
+    public CompletableFuture<Void> requestAllNodeData(BigInteger nodeId) {
+        CompletableFuture<JsonElement> future = sendMessage("nodes", "requestAllData", new Object[] { nodeId });
+        return future.thenAccept(obj -> {
+            // Do nothing, just to complete the future
+        });
+    }
+
+    /**
+     * Request all cluster attribute data for a single endpoint and its children
+     * 
+     * @param nodeId
+     * @param endpointId
+     * @return
+     */
+    public CompletableFuture<Void> requestEndpointData(BigInteger nodeId, Integer endpointId) {
+        CompletableFuture<JsonElement> future = sendMessage("nodes", "requestEndpointData",
+                new Object[] { nodeId, endpointId });
+        return future.thenAccept(obj -> {
+            // Do nothing, just to complete the future
+        });
+    }
+
+    /**
+     * Request a specific cluster attribute data for a single endpoint
+     * 
+     * @param nodeId
+     * @param endpointId
+     * @param clusterId
+     * @return
+     */
+    public CompletableFuture<Void> requestClusterData(BigInteger nodeId, Integer endpointId, Integer clusterId) {
+        CompletableFuture<JsonElement> future = sendMessage("nodes", "requestEndpointData",
+                new Object[] { nodeId, endpointId });
+        return future.thenAccept(obj -> {
+            // Do nothing, just to complete the future
+        });
+    }
+
+    /**
+     * Pair a node using a pairing code, either a manual pairing code or a matter QR code (starts with MT:)
+     * 
+     * @param code
+     * @return
+     */
+    public CompletableFuture<BigInteger> pairNode(String code) {
+        String[] parts = code.trim().split(" ");
+        CompletableFuture<JsonElement> future = null;
+        if (parts.length == 2) {
+            future = sendMessage("nodes", "pairNode", new Object[] { "", parts[0], parts[1] }, PAIRING_TIMEOUT_SECONDS);
+        } else {
+            // MT is a matter QR code, other wise remove any dashes in a manual pairing code
+            String pairCode = parts[0].toUpperCase().indexOf("MT:") == 0 ? parts[0] : parts[0].replaceAll("-", "");
+            future = sendMessage("nodes", "pairNode", new Object[] { pairCode }, PAIRING_TIMEOUT_SECONDS);
+        }
+        return future.thenApply(obj -> {
+            return new BigInteger(obj.getAsString());
+        });
+    }
+
+    /**
+     * Remove a node from the controller
+     * 
+     * @param nodeId
+     * @return
```

Comment:
Add documentation or remove.

Same for the rest of the methods in this class. 😄